File: web/lib/compliance/ad-disclosure.ts
// 광고 표시 문구 강제 삽입 모듈 (3중 방어 중 1차 — 콘텐츠 생성 단계)
// 공정위 표시광고법 준수. 비활성화/우회 절대 금지.

const AD_DISCLOSURE_TEMPLATE = (businessName: string) =>
  `본 포스팅은 [매치블로그]를 통해 ${businessName}으로부터 소정의 원고료를 지원받아 작성되었습니다. #광고 #협찬`

export function injectAdDisclosure(content: string, businessName: string): string {
  const disclosure = AD_DISCLOSURE_TEMPLATE(businessName)
  // 본문 첫 부분에 삽입 (더보기 안에 숨기지 않음)
  return `${disclosure}\n\n${content}`
}

export function hasAdDisclosure(content: string): boolean {
  return (
    content.includes('#광고') &&
    content.includes('#협찬') &&
    content.includes('매치블로그') &&
    content.includes('원고료')
  )
}

File: worker/compliance/ad_disclosure.py
"""
광고 표시 문구 검증 모듈 (3중 방어 중 2차 — 발행 직전)
공정위 표시광고법 준수. 비활성화/우회 절대 금지.
"""

REQUIRED_KEYWORDS = ["#광고", "#협찬", "매치블로그", "원고료"]


def verify_disclosure(content: str) -> bool:
    """발행 직전 광고 문구 포함 여부 최종 확인"""
    return all(kw in content for kw in REQUIRED_KEYWORDS)


def inject_disclosure(content: str, business_name: str) -> str:
    """광고 문구가 없으면 본문 앞에 강제 삽입"""
    if verify_disclosure(content):
        return content
    disclosure = (
        f"본 포스팅은 [매치블로그]를 통해 {business_name}으로부터 "
        f"소정의 원고료를 지원받아 작성되었습니다. #광고 #협찬"
    )
    return f"{disclosure}\n\n{content}"
